[css-grid] Sync WPT CSS Grid 2025-11-18

stwrt opened the pull request:
#### bd68a878214e513a40821fe6a0f3b5ba7caec20f
<pre>
[css-grid] Sync WPT CSS Grid 2025-11-18
<a href="https://bugs.webkit.org/show_bug.cgi?id=302708">https://bugs.webkit.org/show_bug.cgi?id=302708</a>
<a href="https://rdar.apple.com/problem/164959043">rdar://problem/164959043</a>

Reviewed by Tim Nguyen and Sammy Gill.

Sync latest CSS Grid Changes from WPT.

Upstream commit: <a href="https://github.com/web-platform-tests/wpt/commit/f30ddbfca28591661a7db40e84ee8386f0d2e652">https://github.com/web-platform-tests/wpt/commit/f30ddbfca28591661a7db40e84ee8386f0d2e652</a>

* css/css-grid/abspos/grid-abspos-staticpos-align-self-center-large-border-padding.html
* css/css-grid/abspos/grid-abspos-staticpos-align-self-center.html
* css/css-grid/abspos/grid-abspos-staticpos-align-self-end-large-border-padding-ref.html
* css/css-grid/abspos/grid-abspos-staticpos-align-self-flex-end-expected.html
* css/css-grid/abspos/grid-abspos-staticpos-align-self-flex-end-large-border-padding.html
* css/css-grid/abspos/grid-abspos-staticpos-align-self-flex-end-ref.html
* css/css-grid/abspos/grid-abspos-staticpos-align-self-flex-end.html
* css/css-grid/abspos/grid-abspos-staticpos-align-self-self-end-expected.html
* css/css-grid/abspos/grid-abspos-staticpos-align-self-self-end-large-border-padding.html
* css/css-grid/abspos/grid-abspos-staticpos-align-self-self-end-ref.html
* css/css-grid/abspos/grid-abspos-staticpos-align-self-self-end.html
* css/css-grid/abspos/w3c-import.log
* css/css-grid/grid-calc-margin-expected.html
* css/css-grid/grid-calc-margin-ref.html
* css/css-grid/grid-calc-margin.html
* css/css-grid/grid-definition/support/testing-utils.js
* css/css-grid/subgrid/subgrid-item-with-margin-left-auto-expected.html -&gt; css/css-grid/grid-items/grid-size-with-orthogonal-child-001-expected.html
* css/css-grid/grid-items/grid-size-with-orthogonal-child-001.html
* css/css-grid/grid-items/grid-size-with-orthogonal-child-002-expected.html
* css/css-grid/grid-items/grid-size-with-orthogonal-child-002.html
* css/css-grid/grid-items/w3c-import.log
* css/css-grid/grid-model/grid-support-display-001.html
* css/css-grid/layout-algorithm/grid-track-ignores-max-size-001-expected.html
* css/css-grid/layout-algorithm/grid-track-ignores-max-size-001.html
* css/css-grid/layout-algorithm/grid-track-ignores-max-size-002-expected.html
* css/css-grid/grid-items/grid-size-with-orthogonal-child-002.html
* css/css-grid/grid-items/w3c-import.log
* css/css-grid/grid-model/grid-support-display-001.html
* css/css-grid/layout-algorithm/grid-track-ignores-max-size-001-expected.html
* css/css-grid/layout-algorithm/grid-track-ignores-max-size-001.html
* css/css-grid/layout-algorithm/grid-track-ignores-max-size-002-expected.html
* css/css-grid/layout-algorithm/grid-track-ignores-max-size-002.html
* css/css-grid/layout-algorithm/w3c-import.log
* css/css-grid/masonry/tentative/item-placement/w3c-import.log
* css/css-grid/masonry/tentative/parsing/w3c-import.log
* css/css-grid/parsing/grid-template-columns-intrinsic-auto-repeat-computed-implicit-track.tentative.html
* css/css-grid/parsing/grid-template-columns-intrinsic-auto-repeat-computed-nogrid.tentative.html
* css/css-grid/parsing/grid-template-columns-intrinsic-auto-repeat-computed-withcontent.tentative.html
* css/css-grid/parsing/grid-template-columns-intrinsic-auto-repeat-computed.tentative.html
* css/css-grid/parsing/grid-template-columns-invalid.html
* css/css-grid/parsing/grid-template-columns-repeat-auto-valid.tentative-expected.txt
* css/css-grid/parsing/grid-template-columns-repeat-auto-valid.tentative.html
* css/css-grid/parsing/grid-template-columns-repeat-intrinsic-valid.tentative.html
* css/css-grid/parsing/grid-template-rows-intrinsic-auto-repeat-computed-implicit-track.tentative.html
* css/css-grid/parsing/grid-template-rows-intrinsic-auto-repeat-computed-nogrid.tentative.html
* css/css-grid/parsing/grid-template-rows-intrinsic-auto-repeat-computed-withcontent.tentative.html
* css/css-grid/parsing/grid-template-rows-intrinsic-auto-repeat-computed.tentative.html
* css/css-grid/parsing/grid-template-rows-invalid.html
* css/css-grid/parsing/grid-template-rows-repeat-auto-valid.tentative-expected.txt
* css/css-grid/parsing/grid-template-rows-repeat-auto-valid.tentative.html
* css/css-grid/parsing/grid-template-rows-repeat-intrinsic-valid.tentative.html
* css/css-grid/parsing/w3c-import.log
* css/css-grid/subgrid/independent-formatting-context-expected.html
* css/css-grid/subgrid/independent-formatting-context-fieldset-expected.html
* css/css-grid/subgrid/independent-formatting-context-fieldset-ref.html
* css/css-grid/subgrid/independent-formatting-context-fieldset.html
* css/css-grid/subgrid/independent-formatting-context-ref.html
* css/css-grid/subgrid/independent-formatting-context.html
* css/css-grid/subgrid/sticky-subgrid-item-expected.html
* css/css-grid/subgrid/sticky-subgrid-item-ref.html
* css/css-grid/subgrid/sticky-subgrid-item.html
* css/css-grid/subgrid/subgrid-item-with-margin-left-auto-expected.xht
* css/css-grid/subgrid/subgrid-item-with-margin-left-auto.html
* css/css-grid/subgrid/w3c-import.log
* css/css-grid/w3c-import.log

Canonical link: <a href="https://commits.webkit.org/303249@main">https://commits.webkit.org/303249@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/07d0d142c90da98b0f002f41b0570236a359a910

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/131786 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/4278 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/42790 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/139296 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/83660 "Built successfully") | ⏳ 🛠 ios-apple 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/4208 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/4040 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/100727 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/68152 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | ⏳ 🛠 mac-apple 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/134732 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/162/builds/3021 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/118032 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/81514 "Passed tests") | | ⏳ 🛠 vision-apple 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/2908 "Passed tests") | [  ~~🧪 api-mac-debug~~](https://ews-build.webkit.org/#/builders/165/builds/754 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/82517 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/111643 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/36157 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/141941 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/3947 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/36736 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/109100 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/4028 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/3464 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/109263 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/27681 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/2999 "Passed tests") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/114313 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/57157 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/4001 "Built successfully") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/32715 "Passed tests") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/3833 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/4093 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/3961 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

nt1m reviewed (changes requested):
I just realized we should export https://github.com/WebKit/WebKit/commit/44b22eb1df105edd3c9ce77888e752a3138919a7

stwrt committed:
[css-grid] Sync WPT CSS Grid 2025-11-18
https://bugs.webkit.org/show_bug.cgi?id=302708
rdar://problem/164959043

Reviewed by Tim Nguyen and Sammy Gill.

Sync latest CSS Grid Changes from WPT.

Upstream commit: https://github.com/web-platform-tests/wpt/commit/f30ddbfca28591661a7db40e84ee8386f0d2e652

* css/css-grid/abspos/grid-abspos-staticpos-align-self-center-large-border-padding.html
* css/css-grid/abspos/grid-abspos-staticpos-align-self-center.html
* css/css-grid/abspos/grid-abspos-staticpos-align-self-end-large-border-padding-ref.html
* css/css-grid/abspos/grid-abspos-staticpos-align-self-flex-end-expected.html
* css/css-grid/abspos/grid-abspos-staticpos-align-self-flex-end-large-border-padding.html
* css/css-grid/abspos/grid-abspos-staticpos-align-self-flex-end-ref.html
* css/css-grid/abspos/grid-abspos-staticpos-align-self-flex-end.html
* css/css-grid/abspos/grid-abspos-staticpos-align-self-self-end-expected.html
* css/css-grid/abspos/grid-abspos-staticpos-align-self-self-end-large-border-padding.html
* css/css-grid/abspos/grid-abspos-staticpos-align-self-self-end-ref.html
* css/css-grid/abspos/grid-abspos-staticpos-align-self-self-end.html
* css/css-grid/abspos/w3c-import.log
* css/css-grid/grid-calc-margin-expected.html
* css/css-grid/grid-calc-margin-ref.html
* css/css-grid/grid-calc-margin.html
* css/css-grid/grid-definition/support/testing-utils.js
* css/css-grid/subgrid/subgrid-item-with-margin-left-auto-expected.html -> css/css-grid/grid-items/grid-size-with-orthogonal-child-001-expected.html
* css/css-grid/grid-items/grid-size-with-orthogonal-child-001.html
* css/css-grid/grid-items/grid-size-with-orthogonal-child-002-expected.html
* css/css-grid/grid-items/grid-size-with-orthogonal-child-002.html
* css/css-grid/grid-items/w3c-import.log
* css/css-grid/grid-model/grid-support-display-001.html
* css/css-grid/layout-algorithm/grid-track-ignores-max-size-001-expected.html
* css/css-grid/layout-algorithm/grid-track-ignores-max-size-001.html
* css/css-grid/layout-algorithm/grid-track-ignores-max-size-002-expected.html
* css/css-grid/grid-items/grid-size-with-orthogonal-child-002.html
* css/css-grid/grid-items/w3c-import.log
* css/css-grid/grid-model/grid-support-display-001.html
* css/css-grid/layout-algorithm/grid-track-ignores-max-size-001-expected.html
* css/css-grid/layout-algorithm/grid-track-ignores-max-size-001.html
* css/css-grid/layout-algorithm/grid-track-ignores-max-size-002-expected.html
* css/css-grid/layout-algorithm/grid-track-ignores-max-size-002.html
* css/css-grid/layout-algorithm/w3c-import.log
* css/css-grid/masonry/tentative/item-placement/w3c-import.log
* css/css-grid/masonry/tentative/parsing/w3c-import.log
* css/css-grid/parsing/grid-template-columns-intrinsic-auto-repeat-computed-implicit-track.tentative.html
* css/css-grid/parsing/grid-template-columns-intrinsic-auto-repeat-computed-nogrid.tentative.html
* css/css-grid/parsing/grid-template-columns-intrinsic-auto-repeat-computed-withcontent.tentative.html
* css/css-grid/parsing/grid-template-columns-intrinsic-auto-repeat-computed.tentative.html
* css/css-grid/parsing/grid-template-columns-invalid.html
* css/css-grid/parsing/grid-template-columns-repeat-auto-valid.tentative-expected.txt
* css/css-grid/parsing/grid-template-columns-repeat-auto-valid.tentative.html
* css/css-grid/parsing/grid-template-columns-repeat-intrinsic-valid.tentative.html
* css/css-grid/parsing/grid-template-rows-intrinsic-auto-repeat-computed-implicit-track.tentative.html
* css/css-grid/parsing/grid-template-rows-intrinsic-auto-repeat-computed-nogrid.tentative.html
* css/css-grid/parsing/grid-template-rows-intrinsic-auto-repeat-computed-withcontent.tentative.html
* css/css-grid/parsing/grid-template-rows-intrinsic-auto-repeat-computed.tentative.html
* css/css-grid/parsing/grid-template-rows-invalid.html
* css/css-grid/parsing/grid-template-rows-repeat-auto-valid.tentative-expected.txt
* css/css-grid/parsing/grid-template-rows-repeat-auto-valid.tentative.html
* css/css-grid/parsing/grid-template-rows-repeat-intrinsic-valid.tentative.html
* css/css-grid/parsing/w3c-import.log
* css/css-grid/subgrid/independent-formatting-context-expected.html
* css/css-grid/subgrid/independent-formatting-context-fieldset-expected.html
* css/css-grid/subgrid/independent-formatting-context-fieldset-ref.html
* css/css-grid/subgrid/independent-formatting-context-fieldset.html
* css/css-grid/subgrid/independent-formatting-context-ref.html
* css/css-grid/subgrid/independent-formatting-context.html
* css/css-grid/subgrid/sticky-subgrid-item-expected.html
* css/css-grid/subgrid/sticky-subgrid-item-ref.html
* css/css-grid/subgrid/sticky-subgrid-item.html
* css/css-grid/subgrid/subgrid-item-with-margin-left-auto-expected.xht
* css/css-grid/subgrid/subgrid-item-with-margin-left-auto.html
* css/css-grid/subgrid/w3c-import.log
* css/css-grid/w3c-import.log

Canonical link: https://commits.webkit.org/303249@main
